src/Anonymizer/DB.php

```diff
@@ -230,14 +243,22 @@ private function updateData($callback = null): void
             $this->setLimit($this->dbUtils->countResults($this->entity));
         }
 
+        foreach ($this->configuration->getPreQueries() as $preQuery) {
```

Comment:
@edyan they are executed here.

Reply:
if you agree with the previous comment, that action will be replaced by `$expressionLanguage->evaluate($preAction);`